src/Compression/Compression.cs

```diff
@@ -0,0 +1,153 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading.Tasks;
+
+using ObjCRuntime;
+
+namespace Compression {
+
+
+	public partial class CompressionStream : IDisposable {
+
+		[StructLayout(LayoutKind.Sequential)]
+		struct CompressionStreamStruct {
+			public IntPtr Destination;
+			public int DestinationLength;
+			public IntPtr Source;
+			public int SourceLength;
+			public IntPtr State;
+		}
+
+		[DllImport (Constants.libcompression)]
+		static extern int /* CompressionStatus */ CompressionStreamInit (ref CompressionStreamStruct stream, int /* StreamOperation */ operation, int /* CompressionAlgorithm */ algorithm);
```

Comment:
If the size of the `StreamOperation` enum matches the size of the native enum for both 32-bit and 64-bit you can use the enum type in the parameters instead of `int` (same goes for other enums).